pylint/test/unittest_checker_similar.py

```diff
@@ -90,6 +94,51 @@ def test_ignore_imports():
 """.strip()
 
 
+def test_multiline_imports():
+    sys.stdout = StringIO()
```

Comment:
This idiom seems like a perfect candidate for `contextlib. redirect_stdout`

Reply:
Yes, it does.  I'd like to keep this pull request narrowly focused on --ignore-imports, so I sent a separate pull request with this change: https://github.com/PyCQA/pylint/pull/2432